### PR TITLE
Update getting started and architecture for the latest release

### PR DESCRIPTION
**Pull request by grampelberg**
What I need a review of in particular:

- Run through end to end with something fresh and make sure it works (user testing!).
- Any comments on grammar, flow, data, linking to other documentation.

To make it a little easier to review all this, it is up at https://preview.docs.linkerd.io/2/getting-started/.

### CODE REVIEW

File: linkerd.io/content/2/architecture.md [hunk 1]
```diff
@@ -6,19 +6,34 @@ title = "Architecture"
...
+of the data plane. The CLI can be used to
+[install the control plane](../getting-started/).
+
+The control plane is made up of four components:
+
+- Controller - The controller deployment consists multiple containers
+  (public-api, proxy-api, destination, tap) that provide the bulk of the control
+  plane's functionality.
+
+- Web - The web deployment provides the linkerd dashboard.
```
Comment (siggy) [inline linkerd.io/content/2/architecture.md]: `Linkerd`

File: linkerd.io/content/2/architecture.md [hunk 2]
```diff
@@ -27,11 +42,54 @@ as sidecar containers alongside each instance of your service code. In order to
...
 These proxies transparently intercept communication to and from each pod, and
 add features such as instrumentation and encryption (TLS), as well as allowing
 and denying requests according to the relevant policy.
 
 These proxies are not designed to be configured by hand. Rather, their behavior
 is driven by the control plane.
+
+## CLI
+
+The Linkerd CLI is run locally, on your machine and used to interact with the
```
Comment (siggy) [inline linkerd.io/content/2/architecture.md]: maybe drop the comma?

File: linkerd.io/content/2/architecture.md [hunk 3]
```diff
@@ -27,11 +42,54 @@ as sidecar containers alongside each instance of your service code. In order to
...
+control and data planes. It can be used to view statistics, debug production
+issues in real time and install/upgrade the control and data planes.
+
+## Dashboard
+
+The Linkerd Dashboard provides a high level view of what is happening with your
```
Comment (siggy) [inline linkerd.io/content/2/architecture.md]: `Linkerd Dashboard` is capitalized differently elsewhere.

File: linkerd.io/content/2/architecture.md [hunk 4]
```diff
@@ -27,11 +42,54 @@ as sidecar containers alongside each instance of your service code. In order to
...
+
+As a component of the control plane, Grafana provides actionable dashboards for
+your services out of the box. It is possible to see high level metrics and dig
+down into the details, even for pods.
+
+The dashboards that are provided out of the box include:
+
+{{< gallery >}}
+
+{{< gallery-item src="/images/screenshots/grafana-top.png" title="Top Line Metrics" >}}
```
Comment (siggy) [inline linkerd.io/content/2/architecture.md]: are these meant to be clickable? i noticed the source images are big enough they deserve expansion.

File: linkerd.io/content/2/getting-started.md [hunk 5]
```diff
@@ -6,85 +6,79 @@ title = "Getting started"
   weight = 3
 +++
 
-Linkerd has two basic components: a *data plane* comprised of lightweight
-proxies, which are deployed as sidecar containers alongside your service code,
-and a *control plane* of processes that coordinate and manage these proxies.
-Humans interact with the service mesh via a command-line interface (CLI) or
-a web app that you use to control the cluster.
+Linkerd works by installing ultralight [proxies](../architecture#proxy) into
```
Comment (siggy) [inline linkerd.io/content/2/getting-started.md]: `../architecture#proxy` is not a thing?

File: linkerd.io/content/2/getting-started.md [hunk 6]
```diff
@@ -98,269 +92,191 @@ Verify the CLI is installed and running correctly with:
...
+Once piped into `kubectl apply`, Kubernetes will execute a rolling deploy and
+update each pod with the data plane's proxies, all without any downtime.
 
-PODS
-: Lists all pods by requests, success rate, and latency.
+You've added Linkerd to existing services without touching the original YAML!
+Because `inject` augments YAML, it would also be possible to take
+`emojivoto.yml` itself and do the same thing. For a service that isn't a demo,
+it would be ideal to integrate `linkerd inject` into your CI/CD pipeline. This
+allows you to choose what services use linkerd one at a time and incrementally
```
Comment (klingerf) [inline linkerd.io/content/2/getting-started.md]: Typo here: `what services` => `which services`

And: `linkerd` => `Linkerd`

File: linkerd.io/content/2/getting-started.md [hunk 7]
```diff
@@ -98,269 +92,191 @@ Verify the CLI is installed and running correctly with:
...
+roll the data plane across the services that you'd like it on.
```
Comment (klingerf) [inline linkerd.io/content/2/getting-started.md]: "you'd like it on." feels awkward to me -- this whole sentence could use a bit of restructuring.

File: linkerd.io/content/2/getting-started.md [hunk 8]
```diff
@@ -98,269 +92,191 @@ Verify the CLI is installed and running correctly with:
...
 
-REPLICATION CONTROLLER
-: Lists all replications controllers by requests, success rate, and latency.
+Just like with the control plane, it is possible to very that everything worked
```
Comment (klingerf) [inline linkerd.io/content/2/getting-started.md]: Typo here: `very` => `verify`

File: linkerd.io/content/2/getting-started.md [hunk 9]
```diff
@@ -98,269 +92,191 @@ Verify the CLI is installed and running correctly with:
...
-high-level health metrics of deployments in the data plane.
+```bash
+kubectl get -n emojivoto deploy -o yaml \
+  | linkerd inject - \
+  | kubectl apply -f -
+```
 
-NAMESPACE
-: List all the resources in a namespace with requests, success rate and latency.
+This command retrieves all the deployments running in the `emojivoto` namespace,
```
Comment (klingerf) [inline linkerd.io/content/2/getting-started.md]: Typo here: "all the deployments" => "all _of_ the deployments"

File: linkerd.io/content/2/getting-started.md [hunk 10]
```diff
@@ -98,269 +92,191 @@ Verify the CLI is installed and running correctly with:
...
+runs set of Kubernetes resources through `inject`, and finally reapplies it to
```
Comment (klingerf) [inline linkerd.io/content/2/getting-started.md]: Typo here: "runs set" => "runs the set"

File: linkerd.io/content/2/getting-started.md [hunk 11]
```diff
@@ -98,269 +92,191 @@ Verify the CLI is installed and running correctly with:
...
-- Check out the [overview doc](/2/overview)
-- Hop into the #linkerd2 channel on [the Linkerd Slack]
-(https://slack.linkerd.io)
-- Browse through the [Discourse forum](https://discourse.linkerd.io/c/linkerd2).
-- Follow [@linkerd](https://twitter.com/linkerd) on Twitter.
+For more things you can do:
 
-We’re just getting started building Linkerd, and we’re extremely interested in
-your feedback!
+- [Debug emojivoto](../debugging-an-app)
```
Comment (klingerf) [inline linkerd.io/content/2/getting-started.md]: Not a blocker, but this page needs a refresh as well, to include top and fix some of the command output that has change.

File: linkerd.io/content/2/architecture.md [hunk 12]
```diff
@@ -6,19 +6,34 @@ title = "Architecture"
...
 Kubernetes namespace (`linkerd` by default). These services accomplish various
 things---aggregating telemetry data, providing a user-facing API, providing
 control data to the data plane proxies, etc. Together, they drive the behavior
-of the data plane.
+of the data plane. The CLI can be used to
+[install the control plane](../getting-started/).
+
+The control plane is made up of four components:
+
+- Controller - The controller deployment consists multiple containers
```
Comment (klingerf) [inline linkerd.io/content/2/architecture.md]: Typo here: "consists multiple" => "consists _of_ multiple"

File: linkerd.io/content/2/architecture.md [hunk 13]
```diff
@@ -6,19 +6,34 @@ title = "Architecture"
...
+  (public-api, proxy-api, destination, tap) that provide the bulk of the control
+  plane's functionality.
+
+- Web - The web deployment provides the Linkerd dashboard.
+
+- Prometheus - All the metrics exposed by Linkerd are scraped via Prometheus
```
Comment (klingerf) [inline linkerd.io/content/2/architecture.md]: Typo here: "All the metrics" => "All _of_ the metrics"

File: linkerd.io/content/2/architecture.md [hunk 14]
```diff
@@ -27,11 +42,84 @@ as sidecar containers alongside each instance of your service code. In order to
...
 
 These proxies are not designed to be configured by hand. Rather, their behavior
 is driven by the control plane.
+
+### Proxy
+
+An ultralight transparent proxy written in [Rust](https://www.rust-lang.org/),
+the proxy is installed into each pod of a service and becomes part of the data
+plane. It receives all incoming traffic for a pod and intercepts all outgoing
+traffic via. an `initContainer` that configures `iptables` to forward the
```
Comment (klingerf) [inline linkerd.io/content/2/architecture.md]: No "." after "via" here?

File: linkerd.io/content/2/architecture.md [hunk 15]
```diff
@@ -27,11 +42,84 @@ as sidecar containers alongside each instance of your service code. In order to
...
+- Automatic TLS (experimental).
+
+- An on-demand diagnostic tap API.
+
+The proxy supports service discovery via DNS and the
+[destination gRPC API](https://github.com/linkerd/linkerd2-proxy-api).
+
+## CLI
+
+The Linkerd CLI is run locally on your machine and used to interact with the
```
Comment (klingerf) [inline linkerd.io/content/2/architecture.md]: Hmm, maybe "...on your machine and _is_ used to..."

File: linkerd.io/content/2/architecture.md [hunk 16]
```diff
@@ -27,11 +42,84 @@ as sidecar containers alongside each instance of your service code. In order to
...
+
+{{< /gallery >}}
+
+## Prometheus
+
+Prometheus is a cloud native monitoring solution that is used to store all the
+Linkerd metrics. It is installed as part of the control plane and provides the
+data used by the CLI, dashboard and Grafana.
+
+{{< fig src="/images/architecture/metrics.png" title="Metrics Collection" >}}
```
Comment (klingerf) [inline linkerd.io/content/2/architecture.md]: This png is identical to control-plane.png that shows up higher up on the page. I think this one needs to be replaced with something metrics-specific.

File: linkerd.io/content/2/getting-started.md [hunk 17]
```diff
@@ -98,269 +92,191 @@ Verify the CLI is installed and running correctly with:
...
+kubectl -n emojivoto port-forward \
+  $(kubectl -n emojivoto get po -l app=web-svc -oname | cut -d/ -f 2) \
+  8080:80
 ```
 
-Finally, let’s take a look back at our dashboard (run `linkerd dashboard` if you
-haven’t already). You should be able to browse all the services that are running
-as part of the application to view:
+You might notice that some parts of the application are broken! If you were to
+inspect your handly local Kubernetes dashboard, you wouldn’t see very much
```
[the remark this answers is not in the document]
Comment (grampelberg) [reply linkerd.io/content/2/getting-started.md]: Apparently my editor and I need to have a little chat about it deciding not to spell check me anymore!

File: linkerd.io/content/2/getting-started.md (hunk 11 above)
Comment (grampelberg) [reply linkerd.io/content/2/getting-started.md]: Yup, see https://github.com/linkerd/website/pull/24 (still missing screenshots and some polish).

File: linkerd.io/content/2/architecture.md (hunk 14 above)
Comment (grampelberg) [reply linkerd.io/content/2/architecture.md]: I blame @adleong for this.

File: linkerd.io/content/2/architecture.md (hunk 16 above)
Comment (grampelberg) [reply linkerd.io/content/2/architecture.md]: Good call!

**Comment by grampelberg** [comment]
@klingerf thank you for the review! I'm in the process of updating with screenshots and getting those arch slides a little bit more specific.

**Pull request merged**